app/controllers/checkout_controller.rb

```diff
@@ -54,6 +54,9 @@ def update
       else
         update_failed
       end
+    elsif @order.errors.messages[:items_removed].present?
```

Comment:
Yeah it's not great. I couldn't do the redirect to cart page from the model, and simply passing a regular error onto the `update_failed` method at the end made the checkout page freeze up. :/